### DISCUSSION TITLE
feature request: Restart & Run N option (Kernel menu)

### DISCUSSION THREAD

**Discussion by iainjclark** [jupyterlab/jupyterlab#16361] [Ideas]
I have some (many!) workbooks for which I keep restarting the kernel and recomputing cells after making changes to the underlying library. For some of these, Restart & Run All is perfect, but for others, I only want to restart and run up to a particular cell, e.g. [34] - assuming these are in sequence.
My idea: it would be super cool to have a Restart & Run N option which pops up a console window, in which the developer can enter '34' <ENTER> and just let it rip through 34 computes.

**Comment by krassowski** [comment]
That would be pretty similar existing to `Restart Kernel and Run up to Selected Cell`.

![image](https://github.com/jupyterlab/jupyterlab/assets/5832902/b6b25fcd-809a-4256-807e-ba044ed0542a)

 I think this is a reasonable idea, not sure if this should be an extension, or included in core JupyterLab. Maybe it could start as an extension and if it is popular enough it could be merged into the core?

**Reply by iainjclark** [reply]
Many thanks @krassowski for your quick and useful response! I can see that useful Restart Kernel and Run up to Selected Cell option in jupyter labview, but not on treeview. My bad, I should have tried both.

Is there any way to have that option available in treeview (which I prefer, mostly due to the small screen of my laptop)

**Reply by krassowski** [reply]
By treeview do you mean Jupyter Notebook interface? If it is missing there I would open an issue to add it.

**Reply by iainjclark** [reply]
yup the older Jupyter Notebook interface. Will open an issue - many thanks
